pkg/tfbridge/x/token_test.go

```diff
@@ -746,9 +747,98 @@ func TestMaxItemsOneAliasingNested(t *testing.T) {
 	assert.False(t, *info.Resources["pkg_r1"].Fields["f2"].Elem.Fields["n2"].MaxItemsOne)
 }
 
+// (*ProviderInfo).SetAutonaming only applies to fields that are not present in their
+// resource's property map. We need to make sure that unless we mark a field as
+// `MaxItemsOne: nonNil` for some non-nil value, we don't leave that field entry behind
+// since that will disable SetAutonaming.
+func TestMaxItemsOneAliasingWithAutoNaming(t *testing.T) {
```

Comment:
@guineveresaenger @t0yv0 Sorry for the confusing PR. I have added our motivating example as a test case with an explanation of the underlying logic for the change.

Reply:
Brilliant. Oof, so  SetAutonaming does look at whether the SchemaInfo is present or no.

Saw this code:

```go
				if _, hasfield := res.Fields[nameProperty]; !hasfield {
					if res.Fields == nil {
						res.Fields = make(map[string]*SchemaInfo)
					}
					res.Fields[nameProperty] = AutoName(nameProperty, maxLength, separator)
				}
```

I would rephrase

> (*ProviderInfo).SetAutonaming only applies to fields that are not present in their
 // resource's property map.

With 

> (*ProviderInfo).SetAutonaming skips fields that have a SchemaInfo already defined in ResourceInfo.Fields


I always confuse the Info things with he provider maps from shim.SchemaMap.